iceoryx_posh/test/moduletests/test_popo_event_variable.cpp

```diff
@@ -0,0 +1,247 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include "iceoryx_posh/internal/popo/building_blocks/event_listener.hpp"
+#include "iceoryx_posh/internal/popo/building_blocks/event_notifier.hpp"
+#include "iceoryx_posh/internal/popo/building_blocks/event_variable_data.hpp"
+
+#include "test.hpp"
+#include "testutils/timing_test.hpp"
+
+#include <thread>
+
+using namespace ::testing;
+using namespace iox::popo;
+
+class EventVariable_test : public Test
+{
+  public:
+    using notificationVector = iox::cxx::vector<uint64_t, iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET>;
+    EventVariableData m_eventVarData{"Ferdinand"};
+};
+
+TEST_F(EventVariable_test, AllNotificationsAreFalseAfterConstruction)
+{
+    EventVariableData sut;
+    for (auto& notification : sut.m_activeNotifications)
+    {
+        EXPECT_THAT(notification, Eq(false));
+    }
+}
+
+TEST_F(EventVariable_test, CorrectProcessNameAfterConstructionWithProcessName)
+{
+    EXPECT_THAT(m_eventVarData.m_process.c_str(), StrEq("Ferdinand"));
+}
+
+TEST_F(EventVariable_test, AllNotificationsAreFalseAfterConstructionWithProcessName)
+{
+    for (auto& notification : m_eventVarData.m_activeNotifications)
+    {
+        EXPECT_THAT(notification, Eq(false));
+    }
+}
+
+TEST_F(EventVariable_test, NotifyActivatesCorrectIndex)
+{
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 1U;
+    EventNotifier sut(m_eventVarData, index);
+    sut.notify();
+    for (uint64_t i = 0U; i < iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET; i++)
+    {
+        if (i == index)
+        {
+            EXPECT_THAT(m_eventVarData.m_activeNotifications[i], Eq(true));
+        }
+        else
+        {
+            EXPECT_THAT(m_eventVarData.m_activeNotifications[i], Eq(false));
+        }
+    }
+}
+
+TEST_F(EventVariable_test, NotifyActivatesNoIndexIfIndexIsTooLarge)
+{
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET;
+    EventNotifier sut(m_eventVarData, index);
+    sut.notify();
+    for (const auto& notification : m_eventVarData.m_activeNotifications)
+    {
+        EXPECT_THAT(notification, Eq(false));
+    }
+}
+
+TEST_F(EventVariable_test, GetCorrectNotificationVectorAfterNotifyAndWait)
+{
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 1U;
+    EventNotifier notifier(m_eventVarData, index);
+    EventListener listener(m_eventVarData);
+
+    notifier.notify();
+    const auto& activeNotifications = listener.wait();
+    ASSERT_THAT(activeNotifications.size(), Eq(1U));
+    EXPECT_THAT(activeNotifications[0], Eq(index));
+}
+
+TEST_F(EventVariable_test, GetCorrectNotificationVectorAfterMultipleNotifyAndWait)
+{
+    uint64_t index = iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 1U;
+    EventNotifier notifier1(m_eventVarData, index);
+    EventNotifier notifier2(m_eventVarData, 0U);
```

Comment:
please use `FIRST_EVENT_INDEX` and `SECOND_EVENT_INDEX` or something similar

Reply:
done